src/unstable_extensions.rs

```diff
@@ -71,7 +71,7 @@ pub unsafe fn as_chunks_unchecked_mut<const N: usize, T>(this: &mut [T]) -> &mut
 #[inline]
 #[track_caller]
 #[must_use]
-pub fn as_chunks_mut<const N: usize, T>(this: &mut [T]) -> (&mut [[T; N]], &mut [T]) {
+pub fn as_chunks_mut<'a, const N: usize, T>(this: &'a mut [T]) -> (&'a mut [[T; N]], &'a mut [T]) {
```

Comment:
Why do we need this?  `std`'s version doesn't have it, and I'd like to keep the `fn`s in this module identical to `std's implementation (barring things we need to change slightly for privacy and unstability).